src/Tribe/Repositories/Event.php

```diff
@@ -459,37 +459,79 @@ public function filter_by_date_overlaps( $start_datetime, $end_datetime, $timezo
 		global $wpdb;
 		$utc = $this->normal_timezone;
 
-		$lower = Tribe__Date_Utils::build_date_object( $start_datetime, $timezone )->setTimezone( $utc );
-		$upper = Tribe__Date_Utils::build_date_object( $end_datetime, $timezone )->setTimezone( $utc );
-		$lower_string = $lower->format( Tribe__Date_Utils::DBDATETIMEFORMAT );
-		$upper_string = $upper->format( Tribe__Date_Utils::DBDATETIMEFORMAT );
-		$start_key = $this->start_meta_key;
-		$end_key = $this->end_meta_key;
-
+		$lower          = Tribe__Date_Utils::build_date_object( $start_datetime, $timezone )->setTimezone( $utc );
+		$upper          = Tribe__Date_Utils::build_date_object( $end_datetime, $timezone )->setTimezone( $utc );
+		$lower_string   = $lower->format( Tribe__Date_Utils::DBDATETIMEFORMAT );
+		$upper_string   = $upper->format( Tribe__Date_Utils::DBDATETIMEFORMAT );
+		$start_key      = $this->start_meta_key;
+		$end_key        = $this->end_meta_key;
 		$join_start_key = 'tribe_start_date_utc';
-		$join_end_key = 'tribe_end_date_utc';
+		$join_end_key   = 'tribe_end_date_utc';
 
+		// Join on the Event start date, a string in the 'Y-m-d H:i:s' format.
 		$this->filter_query->join(
 			"LEFT JOIN {$wpdb->postmeta} {$join_start_key}
 			ON ( {$wpdb->posts}.ID = {$join_start_key}.post_id
 			AND {$join_start_key}.meta_key = '{$start_key}' )"
 		);
 
+		// Join on the Event end date, a string in the 'Y-m-d H:i:s' format.
 		$this->filter_query->join(
 			"LEFT JOIN {$wpdb->postmeta} {$join_end_key}
 			ON ( {$wpdb->posts}.ID = {$join_end_key}.post_id
 			AND {$join_end_key}.meta_key = '{$end_key}' )"
 		);
 
-		$alt_where = $wpdb->prepare(
-			"(
-				TIMESTAMPDIFF ( SECOND, {$join_start_key}.meta_value, '{$upper_string}' ) >= %d
+		if ( $min_sec_overlap > 1 ) {
+			/*
+			 * If the minimum overlap requested is more than one second, then we have to use a not performant logic.
+			 * The interpolated variables come from the code.
+			 *
+			 * @phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			 */
+			$alt_where = $wpdb->prepare(
+				"(
+				TIMESTAMPDIFF ( SECOND, {$join_start_key}.meta_value, %s ) >= %d
```

Comment:
Why not pass `$join_start_key` (and `$join_end_key`) in as a param as well - quotes?

Reply:
If you mean using the `%i` placeholder, that cannot be used for aliases of tables (I've tried).